### PR TITLE
OEL-1948: Added title and title_tag to patterns.

### PR DESCRIPTION
**Pull request by tibi2303**
Jira issue(s):
- [D8AGE-???](https://webgate.ec.europa.eu/CITnet/jira/browse/D8AGE-???)

### CODE REVIEW

File: tests/src/Kernel/fixtures/markup_rendering_patterns/blockquote.yml [hunk 1]
```diff
@@ -3,41 +3,48 @@ blockquote:
     '#type': pattern
     '#id': blockquote
     '#fields':
+      title: 'The blockqoute title'
```
Comment (brummbar) [inline tests/src/Kernel/fixtures/markup_rendering_patterns/blockquote.yml]: Typo.

File: tests/src/Kernel/fixtures/markup_rendering_patterns/accordion.yml [hunk 2]
```diff
@@ -24,17 +27,19 @@ accordion_with_bare_layout:
       'div.accordion-item div.accordion-collapse.collapse': 3
       'div.accordion-item div.accordion-collapse.collapse.show': 1
     equals:
-      'div.accordion-item:nth-child(1) button.accordion-button': 'Accordion Item #1'
-      'div.accordion-item:nth-child(2) button.accordion-button': 'Accordion Item #2'
-      'div.accordion-item:nth-child(3) button.accordion-button': 'Accordion Item #3'
-      'div.accordion-item:nth-child(1) div.accordion-body': 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium'
-      'div.accordion-item:nth-child(2) div.accordion-body': 'Advice on living, working or travelling in the EU, on visas and immigration for non-EU citizens, European culture. Plus some extra text to make it a little bit longer.'
-      'div.accordion-item:nth-child(3) div.accordion-body': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
+      'h4.bcl-heading': 'The accordion title'
```
Comment (brummbar) [inline tests/src/Kernel/fixtures/markup_rendering_patterns/accordion.yml]: We need to add a different scenario where we don't pass the title, and one where we pass only the title without title tag. You can reduce the number of slides in those scenario. The same should happen for all places where we added the title tag and/or the title.

**Review by brummbar — CHANGES_REQUESTED** [review]
Missing test updates for carousel and ~~content banner~~.

File: templates/patterns/banner/banner.ui_patterns.yml [hunk 3]
```diff
@@ -38,6 +38,10 @@ banner:
       label: 'Title'
       description: 'Title of the banner.'
       preview: 'EU Budget for the future'
+    title_tag:
+      type: text
+      label: 'Title tag'
+      description: 'The tag to use for the title. Defaults to h2.'
```
Comment (drishu) [inline templates/patterns/banner/banner.ui_patterns.yml]:
```suggestion
      description: 'The tag to use for the title. Defaults to div.'
```

File: templates/patterns/accordion/pattern-accordion.html.twig [hunk 4]
```diff
@@ -5,6 +5,8 @@
  */
 #}
 {% include '@oe-bcl/accordion' with {
+  'title': title,
+  'title_tag': title_tag|default('h2'),
```
Comment (drishu) [inline templates/patterns/accordion/pattern-accordion.html.twig]: These defaults are already present in the BCL templates

**Review by drishu — CHANGES_REQUESTED** [review]
I think we should add the title tag also to the card, especially since the default is h1 for some reason, and after we need to check if TwigExtension::bclCardList needs to be adapted so that items passes the new title tag

File: templates/patterns/banner/banner.ui_patterns.yml (hunk 3 above)
Comment (tibi2303) [reply templates/patterns/banner/banner.ui_patterns.yml]: done. works also for card layout.

File: templates/patterns/accordion/pattern-accordion.html.twig (hunk 4 above)
Comment (tibi2303) [reply templates/patterns/accordion/pattern-accordion.html.twig]: done.

File: templates/patterns/card/pattern-card.html.twig [hunk 5]
```diff
@@ -40,6 +40,7 @@
     gutter: 0,
   } : {},
   'title': title,
+  'title_tag': title_tag,
```
Comment (drishu) [inline templates/patterns/card/pattern-card.html.twig:43]: correct me if I'm wrong but I dont see the title tag tested in the kernel pattern test for the card

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/patterns/card/pattern-card.html.twig (hunk 5 above)
Comment (tibi2303) [reply templates/patterns/card/pattern-card.html.twig:43]: done. Also added title_tag for search variant

**Review by AaronGilMartinez — DISMISSED** [review]

File: templates/patterns/featured_media/featured_media.ui_patterns.yml [hunk 6]
```diff
@@ -33,8 +33,8 @@ featured_media:
     title:
       type: text
       label: "Heading"
-      description: "The heading, displayed on the top."
-      preview: "Lorem ipsum"
+      description: 'Title of the featured media.'
+      preview: 'This is the title of this featured media.'
```
Comment (drishu) [inline templates/patterns/featured_media/featured_media.ui_patterns.yml:37]: I missed it the first time around but you can add a title tag here and default it to h2

File: templates/patterns/gallery/pattern-gallery.html.twig [hunk 7]
```diff
@@ -8,7 +8,7 @@
 
 {% set toggle_collapsed = 'view the full gallery (%d)'|t %}
 {% set toggle_expanded = 'view less'|t %}
-{% set title_tag = title_tag|default('h2') %}
+{% set title_tag = title_tag %}
```
Comment (drishu) [inline templates/patterns/gallery/pattern-gallery.html.twig]: this line became redundant

**Review by drishu — CHANGES_REQUESTED** [review]
I missed these 2 points last time;
other than that I did request a oe_whitelabel PR for sanity.

File: templates/patterns/featured_media/featured_media.ui_patterns.yml (hunk 6 above)
Comment (tibi2303) [reply templates/patterns/featured_media/featured_media.ui_patterns.yml:37]: done.

File: templates/patterns/gallery/pattern-gallery.html.twig (hunk 7 above)
Comment (tibi2303) [reply templates/patterns/gallery/pattern-gallery.html.twig]: done.

File: templates/patterns/featured_media/featured_media.ui_patterns.yml (hunk 6 above)
Comment (drishu) [reply templates/patterns/featured_media/featured_media.ui_patterns.yml:37]: you still have to use it in the template, change from `title_tag: 'h2',` to `'title_tag': title_tag|default('h2'),`

**Review by drishu — DISMISSED** [review]

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**